test/Sync/AsyncFileKeyedMutexTest.php

```diff
@@ -0,0 +1,22 @@
+<?php declare(strict_types = 1);
+
+namespace Amp\File\Test\Sync;
+
+use Amp\File\Sync\AsyncFileKeyedMutex;
+use Amp\Sync\KeyedMutex;
+use Amp\Sync\Test\AbstractKeyedMutexTest;
+
+final class AsyncFileKeyedMutexTest extends AbstractKeyedMutexTest
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->ignoreLoopWatchers();
+    }
+
+    public function createMutex(): KeyedMutex
+    {
+        return new AsyncFileKeyedMutex(\tempnam(\sys_get_temp_dir(), 'mutex-') . '%s.lock');
```

Comment:
tempnam shouldn't be used here, as it creates an additional file that isn't used. I guess a bit of random will be fine.